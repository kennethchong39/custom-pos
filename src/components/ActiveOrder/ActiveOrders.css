.active-order {
  overflow: scroll;
  padding: 0 10px;
}

.active-order-header {
  padding-top: 10px;
  text-align: center;
  position: sticky;
  top: 0;
  background-color: white;
  margin-bottom: 3px;
}

.active-order-header h2 {
  margin: 0px;
}
