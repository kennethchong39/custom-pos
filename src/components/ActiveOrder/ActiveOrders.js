import React from 'react';

import Order from '../Order/Order';

import './ActiveOrders.css';

export default function ActiveOrder(props) {
  return (
    <div className="active-order">
      <div className="active-order-header">
        <h2>All Orders</h2>
        <hr />
      </div>
      {props.orders.map((order, index) => {
        return <Order item={order} key={index} index={index} />;
      })}
    </div>
  );
}
