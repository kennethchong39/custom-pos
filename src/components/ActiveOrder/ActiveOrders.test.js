import renderer from 'react-test-renderer';

import ActiveOrders from './ActiveOrders';

test('component: Active Orders tests', () => {
  const testProps = [
    {
      order: [
        {
          name: 'Vegetarian',
          price: 8.99,
          ingredients: {
            bread: 2,
            lettuce: 2,
            tomato: 2,
            cheese: 2,
          },
          quantity: 1,
        },
      ],
      total: 8.99,
    },
  ];
  const tree = renderer.create(<ActiveOrders orders={testProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
