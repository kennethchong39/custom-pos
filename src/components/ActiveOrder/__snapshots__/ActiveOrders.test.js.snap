// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Active Orders tests 1`] = `
<div
  className="active-order"
>
  <div
    className="active-order-header"
  >
    <h2>
      All Orders
    </h2>
    <hr />
  </div>
  <div
    className="order-item"
  >
    <div
      className="order-item-flex"
    >
      <h3>
        Order Number
      </h3>
      <h3>
        # 
        1
      </h3>
    </div>
    <div
      className="order-item-flex-light"
    >
      <p>
        Vegetarian
      </p>
      <p>
        x 
        1
      </p>
      <p>
        $ 
        8.99
      </p>
    </div>
    <div
      className="order-item-flex"
    >
      <p>
        Total
      </p>
      <p>
        $ 
        8.99
      </p>
    </div>
    <button
      className="order-item-button"
      disabled={false}
      onClick={[Function]}
    >
      Complete ?
    </button>
  </div>
</div>
`;
